airflow/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -954,33 +955,49 @@ def find_processing_job_by_name(self, processing_job_name: str) -> bool:
         )
         return bool(self.count_processing_jobs_by_name(processing_job_name))
 
+    @staticmethod
+    def _name_matches_pattern(
+        processing_job_name: str,
+        found_name: str,
+        job_name_suffix: str | None = None,
+    ) -> bool:
+        pattern = re.compile(f"^{processing_job_name}({job_name_suffix})?$")
+        return bool(pattern.search(found_name))
```

Comment:
```suggestion
        return pattern.fullmatch(found_name) is not None
```

Faster since this is a full string match.

Reply:
Nice, thanks.